### DISCUSSION TITLE
hbbs and hbbr key location or key is not found when using the /full/path/to/hbbs executable

### DISCUSSION THREAD

**Discussion by jlinkels** [rustdesk/rustdesk-server#484] [General]
The hbbs public and private keys are generated and stored in the same directory as the hbbs executable.
When I run ./hbbs from that same directory all is fine.

However I created a systemd script for starting and stopping hbbs. Installation directory /opt/rustdesk. Systemd handles this fine, but apparently the key is not loaded in hbbs. I get a key error.

When I add the public key to the startup command: /opt/rustdesk/hbbs -k 'thisisthepublickeyformyinstallation' all is fine. 

Since this is the public key it does not compromise security. Still it is not very elegant that the key is not taken from the executable directory. When I change the key pair I must remember to update the script.

What am I missing here or what am I doing wrong?

**Comment by MatrixUser** [comment]
I am not familiar with systemd scripts but you may have to to include
**cd /opt/rustdesk**
in your script before starting the hbbs and the hbbr binaries. I am running the rustdesk-server on an Arm v7 based ZGemma H7C STB and a Fritz!Box 7690, which both have and embedded linux firmwares without systemd, and my script to start the server looks like that:

#!/bin/sh
cd /opt/rustdesk
/opt/rustdesk/hbbs -k _ > /opt/rustdesk/log/signalserver.log 2> /opt/rustdesk/log/signalserver.error &
/opt/rustdesk/hbbr -k _ > /opt/rustdesk/log/relayserver.log 2> /opt/rustdesk/log/relayserver.error &

Without the cd command, hbbs and hbbr are not able to pick up the key files.

**Reply by jlinkels** [reply]
You are right. However the **cd** command does not exist for systemd scripts, but the **WorkingDirectory** directive exists. I added that and it seems to work.

I did a daemon-reload and then restart the services. RustDesk is still on-line.

So thank you.
